internal/exporter/exporter.go

```diff
@@ -0,0 +1,168 @@
+package exporter
+
+import (
+	"archive/tar"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"github.com/ethersphere/bee/pkg/shed"
+	"io"
+	"os"
+)
+
+const (
+	// filename in tar archive that holds the information
+	// about exported data format version
+	ExportVersionFilename = ".swarm-export-version"
+	// current export format version
+	CurrentExportVersion = "1"
+	// default export filename
+	DefaultExportFilename = "swarm-exportdb.tar"
+)
+
+type ProgressUpdater interface {
+	Update(int, int)
+}
+
+type Option func(*exporter)
+
+func WithDestinationFilename(fname string) Option {
+	return func(e *exporter) {
+		e.dstFile = fname
+	}
+}
+
+func WithProgressUpdater(upd ProgressUpdater) Option {
+	return func(e *exporter) {
+		e.updater = upd
+	}
+}
+
+func Export(src string, opts ...Option) error {
+	e, err := newExporter(src, opts...)
+	if err != nil {
+		return errors.New("Invalid source directory err:" + err.Error())
```

Comment:
Error messages should not be capitalized https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Use fmt.Errorf with %w verb to propagate error type instead only its string.